cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CXXLanguageFrontend.kt

```diff
@@ -151,14 +151,13 @@ class CXXLanguageFrontend(
                 if (topLevel != null) {
```

Comment:
This whole method seems to be very complicated to check if the `path` is contained in the `list`. Why isn't it something like `list.any{Files.isSameFile(it, path)}`?

Reply:
We basically need to check if any of the includeLocations + path is in list. Not sure if this can be achieved any easier.

Reply:
Maybe `includeLocations.any { include -> list.any { Files.isSameFile(it, include.relativize(path)) } }` could work

Reply:
Nope, doesn't since relative can throw an error (which we ignore) if the paths are not relative to each other :(